src/obstacle_params_parser.cpp

```diff
@@ -0,0 +1,172 @@
+            new hpp::fcl::Box(x, y, z));
+      } else {
+        std::cerr << "No dimension or wrong dimensions in the obstacle "
+                     "config. Try to use "
+                     "the ones for the shapes desired, such as 'radius' for "
+                     "'sphere','x', 'y', 'z' for 'box' or 'halfLength' and "
+                     "'radius' for 'capsule'."
+                  << std::endl;
+        return;
+      }
+
+    } else if (type == "capsule") {
+      double radius;
+      double halfLength;
+      if (pnh_->hasParam(obstacle_name + "/radius") &&
+          pnh_->hasParam(obstacle_name + "/halfLength")) {
+        pnh_->getParam(obstacle_name + "/radius", radius);
+        pnh_->getParam(obstacle_name + "/halfLength", halfLength);
+        geometry = pinocchio::GeometryObject::CollisionGeometryPtr(
+            new hpp::fcl::Capsule(radius, halfLength));
+      }
+    } else {
+      std::cerr << "No type or wrong type in the obstacle config. Try to use "
+                   "the one implemented, such as 'sphere', 'box' or 'capsule'."
+                << std::endl;
+      return;
+    }
+    pinocchio::SE3 obstacle_pose(rotation, translation);
+    pinocchio::GeometryObject obstacle(obstacle_name, 0, 0, geometry,
+                                       obstacle_pose);
+    collision_model_->addGeometryObject(obstacle);
+    obs_idx += 1;
+  }
+  // Adding the collision pairs to the geometry model
+
+  XmlRpc::XmlRpcValue collision_pairs;
+
+  if (pnh_->getParam("collision_pairs", collision_pairs)) {
+    if (collision_pairs.getType() == XmlRpc::XmlRpcValue::TypeArray &&
+        collision_pairs.size() > 0) {
+      for (int i = 0; i < collision_pairs.size(); ++i) {
+        if (collision_pairs[i].getType() == XmlRpc::XmlRpcValue::TypeArray &&
+            collision_pairs[i].size() == 2) {
+          if (collision_pairs[i][0].getType() ==
+                  XmlRpc::XmlRpcValue::TypeString &&
+              collision_pairs[i][1].getType() ==
+                  XmlRpc::XmlRpcValue::TypeString) {
+            std::string object1 =
+                static_cast<std::string>(collision_pairs[i][0]);
+            std::string object2 =
+                static_cast<std::string>(collision_pairs[i][1]);
+
+            addCollisionPair(object1, object2);
+          } else {
+            std::cerr << "Invalid collision pair type." << std::endl;
+            return;
+          }
+        } else {
+          std::cerr << "Invalid collision pair number." << std::endl;
+          return;
+        }
+      }
+    } else {
+      std::cerr << "No collision pair." << std::endl;
+      return;
+    }
+  }
+}
+
+void ObstacleParamsParser::addCollisionPair(const std::string &name_object1,
+                                            const std::string &name_object2) {
+  std::size_t object1Id = collision_model_->getGeometryId(name_object1);
+  std::size_t object2Id = collision_model_->getGeometryId(name_object2);
```

Comment:
```suggestion
  const std::size_t object1Id = collision_model_->getGeometryId(name_object1);
  const std::size_t object2Id = collision_model_->getGeometryId(name_object2);
```
Those values don't change later in this function so they can be const